### PR TITLE
Added test to check if simulation has been solved in `simulation.py` and `batch_study.py`

### LINKED ISSUE #3725: Check if the solution exists before creating a GIF (`Simulation.create_gif` and `BatchStudy.create_gif`)
I liked how #3723 checks if the simulations have been solved before proceeding ahead to the plotting functionality. It would be nice to have similar checks in -

https://github.com/pybamm-team/PyBaMM/blob/c3def31a04df163769851733657d9b49b3700bca/pybamm/batch_study.py#L184-L187

and 

https://github.com/pybamm-team/PyBaMM/blob/c3def31a04df163769851733657d9b49b3700bca/pybamm/simulation.py#L1047-L1050

### PR DESCRIPTION
**Pull request by prady0t**
# Description

Simple check for if simulations have been solved or not.

Fixes #3725 

## Type of change

Please add a line in the relevant section of [CHANGELOG.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CHANGELOG.md) to document the change (include PR #) - note reverse order of PR #s. If necessary, also add to the list of breaking changes.

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [ ] No style issues: `$ pre-commit run` (or `$ nox -s pre-commit`) (see [CONTRIBUTING.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CONTRIBUTING.md#installing-and-using-pre-commit) for how to set this up to run automatically when committing locally, in just two lines of code)
- [ ] All tests pass: `$ python run-tests.py --all` (or `$ nox -s tests`)
- [ ] The documentation builds: `$ python run-tests.py --doctest` (or `$ nox -s doctests`)

You can run integration tests, unit tests, and doctests together at once, using `$ python run-tests.py --quick` (or `$ nox -s quick`).

## Further checks:

- [ ] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added that prove fix is effective or that feature works

### CODE REVIEW

File: pybamm/batch_study.py [hunk 1]
```diff
@@ -196,7 +197,8 @@ def create_gif(self, number_of_images=80, duration=0.1, output_filename="plot.gi
             Name of the generated GIF file.
 
         """
-
+        if not self.sims:
```
Comment (Saransh-cpp) [inline pybamm/batch_study.py:200]: You could do something like -
```suggestion
        if not hasattr(self, "sims"):
```
I think this should work.

File: pybamm/batch_study.py [hunk 2]
```diff
@@ -86,6 +86,7 @@ def __init__(
         self.repeats = repeats
         self.permutations = permutations
         self.quick_plot = None
+        self.sims = []
```
Comment (Saransh-cpp) [inline pybamm/batch_study.py:89]: Initializing this in `__init__` would be redundant. We can keep this in its original place.

**Review by Saransh-cpp — CHANGES_REQUESTED** [review]
Thanks, @prady0t! Looks really nice! A couple of comments below -

File: pybamm/batch_study.py (hunk 2 above)
Comment (Saransh-cpp) [reply pybamm/batch_study.py:89]: and using `hasattr` will still work

**Comment by prady0t** [comment]
@Saransh-cpp I've done the suggested changes here.

**Review by Saransh-cpp — APPROVED** [review]
Thanks, @prady0t! Looks good to me!

**Comment by agriyakhetarpal** [comment]
Should we add a CHANGELOG entry here?

**Comment by kratman** [comment]
> Should we add a CHANGELOG entry here?

I think it is a small enough change that it does not need it. However that one test keeps failing and I doubt it is due to this small PR. Should we open a PR for that or do you already know what is happening?

**Comment by agriyakhetarpal** [comment]
> Should we open a PR for that or do you already know what is happening?

No worries, it looks like it was coming from a cached wheel for `scikits.odes` – I deleted the caches and the test passed this time. Let's merge

**Comment by agriyakhetarpal** [comment]
@all-contributors please add @prady0t for code, test

**Pull request merged**